server/src/main/java/org/elasticsearch/index/mapper/ParsedDocument.java

```diff
@@ -35,6 +39,54 @@
 
     private Mapping dynamicMappingsUpdate;
 
+    /**
+     * Create a no-op tombstone document
+     * @param reason    the reason for the no-op
+     */
+    public static ParsedDocument noopTombstone(String reason) {
+        Document document = new Document();
+        SeqNoFieldMapper.SequenceIDFields seqIdFields = SeqNoFieldMapper.SequenceIDFields.tombstone();
+        seqIdFields.addFields(document);
+        Field versionField = VersionFieldMapper.versionField();
+        document.add(versionField);
+        // Store the reason of a noop as a raw string in the _source field
+        final BytesRef byteRef = new BytesRef(reason);
+        document.add(new StoredField(SourceFieldMapper.NAME, byteRef.bytes, byteRef.offset, byteRef.length));
+        return new ParsedDocument(
+            versionField,
+            seqIdFields,
+            "",
+            null,
+            Collections.singletonList(document),
+            new BytesArray("{}"),
```

Comment:
it looks odd that the source may not be {} but it always is here, should we rather provide the same bytesref as above to the method? I don't know if this has implications.

Reply:
This is how it worked before, I think the code path for tombstone docs never actually calls `source()` on the ParsedDocument.

Reply:
I see, I had not noticed. Maybe pass in the same bytes ref for consistency, and you end up creating one less bytes ref instance too :)

Reply:
I confused the BytesRef above used for the source stored field, with this bytes array used as source. If this is never used, shall we pass null instead? Or if you are nervous about changing it, leave it as-is :)

Reply:
I think I'd prefer to leave it...